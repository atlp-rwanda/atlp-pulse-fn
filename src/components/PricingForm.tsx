import React from 'react';
import { Icon } from '@iconify/react';

export default function PricingForm() {
  return (
    <div className="flex flex-col bg-light-bg dark:bg-dark-frame-bg justify-center mx-auto w-full py-32">
      <h2 className="text-light-text dark:text-dark-text-fill text-4xl font-bold mb-4 mx-auto">
        Pro
      </h2>
      <h3 className="text-light-text dark:text-dark-text-fill text-3xl mb-4 mx-auto">
        Fill this form
      </h3>
      <div className="w-3/4 md:w-2/4 mx-auto">
        <form>
          <div className="mb-4">
            <div className="flex flex-row items-center">
              <span>
                <Icon icon="codicon:organization" color="#148fb6" />
              </span>
              <h4 className="text-light-text dark:text-dark-text-fill text-xl ml-2">
                Organization
              </h4>
            </div>
            <div className="my-2 flex flex-col">
              <label className="text-light-text dark:text-dark-text-fill text-lg">
                Country
              </label>
              {/* <input type="text" placeholder="Country"></input> */}
              <input
                className="dark:bg-dark-frame-bg appearance-none border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
                id="country"
                type="text"
                placeholder="Country"
              ></input>
            </div>
            <div>
              <label className="text-light-text dark:text-dark-text-fill text-lg">
                Organization Name
              </label>
              {/* <input type="text" placeholder="Organization Name"></input> */}
              <input
                className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
                id="organizationName"
                type="text"
                placeholder="Organization name"
              ></input>
            </div>
            <div>
              <div>
                <label className="text-light text dark:text-dark-text-fill text-lg">
                  Street address
                </label>
                {/* <input type="text" placeholder="Street address"></input> */}
                <input
                  className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
                  id="streetAddress"
                  type="text"
                  placeholder="Street address"
                ></input>
              </div>
              <div>
                <label className="text-light-text dark:text-dark-text-fill text-lg">
                  City
                </label>
                {/* <input type="text" placeholder="City"></input> */}
                <input
                  className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
                  id="city"
                  type="text"
                  placeholder="City"
                ></input>
              </div>
            </div>
            <div>
              <div>
                <label className="text-light-text dark:text-dark-text-fill text-lg">
                  Province
                </label>
                {/* <input type="text" placeholder="Province"></input> */}
                <input
                  className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
                  id="province"
                  type="text"
                  placeholder="Province"
                ></input>
              </div>
              <div>
                <label className="text-light-text dark:text-dark-text-fill text-lg">
                  Postal Code
                </label>
                {/* <input type="text" placeholder="Postal Code"></input> */}
                <input
                  className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
                  id="pcode"
                  type="text"
                  placeholder="Postal code"
                ></input>
              </div>
            </div>
          </div>
          <div className="mb-4">
            <div className="flex flex-row items-center">
              <span>
                <Icon icon="fluent:payment-32-filled" color="#148fb6" />
              </span>
              <h4 className="text-light-text dark:text-dark-text-fill text-xl ml-2">
                Payment method
              </h4>
            </div>
            <div>
              <label className="text-light-text dark:text-dark-text-fill text-lg">
                Name
              </label>
              {/* <input type="text" placeholder="Name"></input> */}
              <input
                className="appearance-none border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill dark:bg-dark-frame-bg leading-tight focus:outline-none focus:shadow-outline"
                id="name"
                type="text"
                placeholder="Name"
              ></input>
            </div>
            <div>
              <div>
                <label className="text-light-text dark:text-dark-text-fill text-lg">
                  Card number
                </label>
                {/* <input type="text" placeholder="Card nbumber"></input> */}
                <input
                  className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
                  id="cardNumber"
                  type="text"
                  placeholder="Card number"
                ></input>
              </div>
              <div>
                <label className="text-light-text dark:text-dark-text-fill text-lg">
                  Expiration
                </label>
                {/* <input type="text" placeholder="Expiration"></input> */}
                <input
                  className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
                  id="expiration"
                  type="text"
                  placeholder="Expiration"
                ></input>
              </div>
            </div>
            <div>
              <div>
                <label className="text-light-text dark:text-dark-text-fill text-lg">
                  CVV
                </label>
                <input
                  className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
                  id="cvv"
                  type="text"
                  placeholder="CVV"
                ></input>
              </div>
              <div>
                <label className="text-light-text dark:text-dark-text-fill text-lg ">
                  Postal Code
                </label>
                <input
                  className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
                  id="pcode"
                  type="text"
                  placeholder="Postal code"
                ></input>
              </div>
            </div>
          </div>
          <button className="bg-cyan-600 hover:bg-cyan-700 text-white py-2 px-4 mb-2 rounded">
            Request
          </button>
        </form>
      </div>
    </div>
  );
}
