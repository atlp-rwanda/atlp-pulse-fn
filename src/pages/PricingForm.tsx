import React from 'react';
import PricingForm from '../components/PricingForm';

export default function PricingForm() {
  return (
    <div className="bg-light-bg dark:bg-dark-frame-bg flex flex-col justify-center items-center mx-auto w-2/4 pt-10">
      <div className="flex flex-col dark:bg-dark-frame-bg justify-center m-10">
        <PricingForm />
      </div>
    </div>
  );
}
