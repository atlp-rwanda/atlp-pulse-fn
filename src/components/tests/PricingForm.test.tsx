import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import '../../../test/jest/__mocks__/matchMedia';
import PricingForm from '../PricingForm';

describe('Pricing form  test ', () => {
  it('Should render Pricing form and payment form', () => {
    const elem = renderer
      .create(
        <BrowserRouter>
          <PricingForm />
        </BrowserRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
