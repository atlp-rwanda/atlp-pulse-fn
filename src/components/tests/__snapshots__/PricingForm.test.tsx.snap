// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pricing form  test  Should render Pricing form and payment form 1`] = `
<div
  className="flex flex-col bg-light-bg dark:bg-dark-frame-bg justify-center mx-auto w-full py-32"
>
  <h2
    className="text-light-text dark:text-dark-text-fill text-4xl font-bold mb-4 mx-auto"
  >
    Pro
  </h2>
  <h3
    className="text-light-text dark:text-dark-text-fill text-3xl mb-4 mx-auto"
  >
    Fill this form
  </h3>
  <div
    className="w-3/4 md:w-2/4 mx-auto sm:w-full  pl-32"
  >
    <form
      className="items-center"
    >
      <div
        className="mb-4"
      >
        <div
          className="flex flex-row items-center"
        >
          <span>
            <span />
          </span>
          <h4
            className="text-light-text dark:text-dark-text-fill text-xl ml-2"
          >
            Organization
          </h4>
        </div>
        <div
          className="my-2 flex flex-col"
        >
          <label
            className="text-light-text dark:text-dark-text-fill text-lg"
          >
            Country
          </label>
          <select
            className="dark:bg-dark-frame-bg  appearance-none border my-1 rounded w-3/4 py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
          >
            <option
              selected={true}
            >
              Select a country
            </option>
            <option
              value="US"
            >
              United States
            </option>
            <option
              value="Rw"
            >
              Rwanda
            </option>
            <option
              value="BU"
            >
              Burundi
            </option>
            <option
              value="TZD"
            >
              Tanzania
            </option>
            <option
              value="NGR"
            >
              Nigeria
            </option>
          </select>
        </div>
        <div
          className="flex flex-col"
        >
          <label
            className="text-light-text dark:text-dark-text-fill text-lg"
          >
            Organization Name
          </label>
          <input
            className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-3/4 py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
            id="organizationName"
            placeholder="Organization name"
            type="text"
          />
        </div>
        <div
          className=" flex justify-between align-baseline flex-wrap pl-0 py-2 w-3/4"
        >
          <div
            className="flex flex-col"
          >
            <label
              className="text-light text dark:text-dark-text-fill text-lg"
            >
              Street address
            </label>
            <input
              className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
              id="streetAddress"
              placeholder="Street address"
              type="text"
            />
          </div>
          <div
            className="flex justify-between flex-col"
          >
            <label
              className="text-light-text dark:text-dark-text-fill text-lg"
            >
              City
            </label>
            <input
              className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
              id="city"
              placeholder="City"
              type="text"
            />
          </div>
        </div>
        <div
          className="flex justify-between  flex-wrap pl-0 py-2 w-3/4"
        >
          <div
            className="flex flex-col"
          >
            <label
              className="text-light-text dark:text-dark-text-fill text-lg"
            >
              Province
            </label>
            <input
              className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
              id="province"
              placeholder="Province"
              type="text"
            />
          </div>
          <div
            className="flex flex-col"
          >
            <label
              className="text-light-text dark:text-dark-text-fill text-lg"
            >
              Postal Code
            </label>
            <input
              className="appearance-none dark:bg-dark-frame-bg border my-1 rounded w-full py-2 px-3 text-light-text dark:text-dark-text-fill leading-tight focus:outline-none focus:shadow-outline"
              id="pcode"
              placeholder="Postal code"
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="mb-4"
      >
        <div
          className="flex flex-row items-center"
        >
          <span>
            <span />
          </span>
          <h4
            className="text-light-text dark:text-dark-text-fill text-xl ml-2"
          >
            Payment method
          </h4>
        </div>
        <div />
        <div />
      </div>
      <div
        className="flex flex-row  flex-wrap justify-between w-3/4"
      >
        <div>
          <span />
          <h3
            className="text-light-text dark:text-dark-text-fill sm:text-sm"
          >
            Credit Card
          </h3>
        </div>
        <a
          href="/redirect-info"
          onClick={[Function]}
        >
          <div>
            <span />
            <h3
              className="text-light-text dark:text-dark-text-fill sm:text-sm"
            >
              Apple Pay
            </h3>
          </div>
        </a>
        <a
          href="/redirect-info"
          onClick={[Function]}
        >
          <div>
            <span />
            <h3
              className="text-light-text dark:text-dark-text-fill sm:text-sm"
            >
              PayPal
            </h3>
          </div>
        </a>
      </div>
    </form>
  </div>
</div>
`;
